packages/block-library/src/query/edit/query-placeholder.js

```diff
@@ -54,7 +54,7 @@ export default function QueryPlaceholder( {
 		matchingVariation?.icon ||
 		blockType?.icon?.src;
 	const label = matchingVariation?.title || blockType?.title;
-	if ( isStartingBlank ) {
+	if ( ! hasPatterns || isStartingBlank ) {
```

Comment:
With this change the variation picker is shown initially until the `getPatternsByBlockTypes` resolves. Is this what we want?

Reply:
What would a good alternative be here in terms of UI? We could do the opposite and show the variation picker by default until it is resolved? I'm not sure that is better though. We could show a spinner?

Reply:
Maybe show the placeholder as is(that means show the button), and have the `hasPatterns` in a `useEffect` to `isStartingBlank:true`? 🤔 It's far more common to have patterns so IMO it makes sense to do it that way. What do you think?

Reply:
Yes makes sense! Will fix this up. 👍

Reply:
Thank you and sorry for the back and forth here.. It seems this is way more convoluted as all the pattern selector calls etc.. always return an `empty array`. This means we'll always show the variation picker first and then show the placeholder if we do have patterns. 

I couldn't find any other way currently to also check if they have loaded first(https://github.com/WordPress/gutenberg/blob/trunk/packages/core-data/src/resolvers.js#L506). And I even tried adding a new prop in state like:
```
export function blockPatternsHaveResolved( state = false, action ) {
	switch ( action.type ) {
		case 'RECEIVE_BLOCK_PATTERNS':
			return true;
	}
	return state;
}
```

But that didn't work as expected too, because we still pass the patterns in editor settings(`__experimentalBlockPatterns`) and the update to that is async. That means the patterns are loaded, but the setting is updated afterwards and still has an empty array as a default value.

Maybe the best way forward would be to eventually remove the block patterns as a setting and introduce a new `fetch` function for them in editor settings. It has been discussed before for the `reusable` blocks as well..

Maybe there is another way I haven't thought of, but I think we should rush a PR, that renders initially the variation picker when the most common case is for patterns to exist..

--cc @jsnajdr